optimum/habana/transformers/trainer.py

```diff
@@ -729,7 +717,7 @@ def hpu_deepspeed_checkpointing(function, *checkpoint_args):
 
         # set_to_none is not implemented for some optimizers
         try:
-            model.zero_grad(set_to_none=True)
+            model.zero_grad(set_to_none=False)
```

Comment:
set_to_none=False is unconditional, as all_reduce_gradients depends on fixed tensors.

Reply:
Maybe we can keep `set_to_none=True` when the world size is 1?